itest/psbt_test.go

```diff
@@ -1610,6 +1618,352 @@ func testPsbtSighashNoneInvalid(t *harnessTest) {
 	require.ErrorContains(t.t, err, "unable to verify proof")
 }
 
+// testPsbtTrustlessSwap tests that the SIGHASH_NONE flag of vPSBTs can be used
+// to execute a trustless swap between two parties. This is done by using
+// different sighashes for the bitcoin psbt and taproot asset vpsbt. One is able
+// to "claim" the assets only by bringing their own bitcoin to fulfill the
+// outputs of the bitcoin transaction.
+func testPsbtTrustlessSwap(t *harnessTest) {
+	// First, we'll make a normal asset.
+	rpcAssets := MintAssetsConfirmBatch(
+		t.t, t.lndHarness.Miner.Client, t.tapd,
+		[]*mintrpc.MintAssetRequest{issuableAssets[0]},
+	)
+
+	mintedAsset := rpcAssets[0]
+	genInfo := mintedAsset.AssetGenesis
+
+	ctxb := context.Background()
+
+	var (
+		alice       = t.tapd
+		numUnits    = mintedAsset.Amount
+		chainParams = &address.RegressionNetTap
+		assetID     asset.ID
+	)
+	copy(assetID[:], genInfo.AssetId)
+
+	// Now Alice will derive the script and anchor internal keys that will
+	// be used to bootstrap an interactive full send of her assets.
+	aliceScriptKey, aliceAnchorInternalKey := DeriveKeys(t.t, alice)
+	vPkt := tappsbt.ForInteractiveSend(
+		assetID, numUnits, aliceScriptKey, 1, aliceAnchorInternalKey,
```

Comment:
For clarify we could maybe have `aliceScriptKey` just be a blank pub key? In that later on Bob will swap this out in order to claim this himself.

Reply:
We'll need to derive the anchor key too so I guess this is less lines overall?

Will rename the variable to describe intent